ext/opcache/Optimizer/zend_cfg.c

```diff
@@ -557,6 +562,7 @@ int zend_build_cfg(zend_arena **arena, const zend_op_array *op_array, uint32_t b
 				break;
 			case ZEND_SWITCH_LONG:
 			case ZEND_SWITCH_STRING:
+			case ZEND_MATCH:
```

Comment:
This shouldn't assign `j + 1` as a successor. MATCH never falls through, right?

Reply:
Hm... Doing that incorrectly optimizes the following code (but only when also adjusting the `successors_count`):

```php
var_dump(match(true) {
    1, 2, 3, 4, 5 => 'foo',
});

// Before
// 0000 INIT_FCALL 1 96 string("var_dump")
// 0001 V1 = NEW 0 string("UnhandledMatchError")
// 0002 DO_FCALL
// 0003 THROW V1

// After
// 0000 INIT_FCALL 1 96 string("var_dump")
// 0001 SEND_VAL string("foo") 1
// 0002 DO_ICALL
// 0003 JMP 0009
```

What I did is:

```c
block->successors_count = (opline->opcode == ZEND_MATCH ? 1 : 2) + zend_hash_num_elements(jumptable);
...
if (opline->opcode != ZEND_MATCH) {
	block->successors[s++] = j + 1;
}
```

Also had to change an assertion in block_pass.c. Not sure what's happening here, I'll investigate tomorrow.